presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -219,5 +220,11 @@ public PlanNode visitProject(ProjectNode node, RewriteContext<PlanNode> context)
 
             return new ProjectNode(idAllocator.getNextId(), rewrittenNode.getSource(), assignments);
         }
+
+        @Override
+        public PlanNode visitGroupReference(GroupReference node, RewriteContext<PlanNode> context)
```

Comment:
It is strange that we don't resolve `GroupReferences` in `ExtendProjectionRewriter` while other visitors do resolve `GroupReferences`. It's feels fishy. Is that correct behaviour @kokosing?

Reply:
It seems that all other visitors are not called from rules, are they?

Reply:
@sopel39 i retained the existing behavior. I didn't understand that though